pt_miniscreen/hotspots/base.py

```diff
@@ -0,0 +1,95 @@
+from dataclasses import dataclass
+from time import perf_counter
+
+from PIL import Image, ImageOps
+from pitop.miniscreen.oled.assistant import MiniscreenAssistant
+
+from ..types import Coordinate
+
+
+class Hotspot:
+    def __init__(self, interval, size):
+        self.interval = interval
+        self.size = size
+
+        self.draw_white = True
+        self.draw_black = False
+
+        self.last_updated = -self.interval
+        self.invert = False
+        self.width = size[0]
+        self.height = size[1]
+
+    @property
+    def width(self):
+        return self.size[0]
+
+    @width.setter
+    def width(self, value):
+        self.size = (value, self.height)
+
+    @property
+    def height(self):
+        return self.size[1]
+
+    @height.setter
+    def height(self, value):
+        self.size = (self.width, value)
+
+    def should_redraw(self):
+        if not self.draw_white and not self.draw_black:
+            return False
+
+        """Only requests a redraw after ``interval`` seconds have elapsed."""
+        return perf_counter() - self.last_updated > self.interval
+
+    @property
+    def image(self):
```

Comment:
Does this get used? I think we use `paste_into` instead. I think `last_updated` might not be used either